test/topology_experimental_raft/test_tombstone_gc.py

```diff
@@ -44,3 +50,84 @@ async def test_default_tombstone_gc_does_not_override(manager: ManagerClient, rf
         async with new_test_table(cql, keyspace, "p int primary key, x int", " with tombstone_gc = {'mode': 'disabled'}") as table:
             await cql.run_async(f"ALTER TABLE {table} add y int")
             check_tombstone_gc_mode(cql, table, "disabled")
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail(reason="issue #15607")
+async def test_group0_tombstone_gc(manager: ManagerClient):
+    """
+    Regression test for #15607.
+
+    Test that the tombstones are being cleaned after the group 0 catching up.
+
+    Arrange:
+        - start 3 nodes
+    Act:
+        - create new group0 tombstones by updating the schema (create/alter/delete randome tables)
+    Assert:
+        - the tombstones are cleaned up eventually
+    """
+    cmdline = [
+        '--enable-cache', '0',
```

Comment:
Added a comment for this - basically, the tombstones remain in the cache otherwise, so the alternative is to drop caches after the compaction or filter the cached tombstones out after the "select from mutation_fragments".

I've seen this used in some other tests so it seems to be the easiest to just disable the caches.

Reply:
@denesb is the mutation compacting logic working correctly then? The tombstones are expired, but `select ... from mutation_fragments` is still showing them. Is that expected?

Reply:
@kbr-scylla compaction happens when populating the cache from the sstable set, and expired tombstones that were already in the cache will be purged on reads, but let's remember not all tombstone types are handled today, see: https://github.com/scylladb/scylladb/issues/16092. what about doing the validation with bypass cache instead? ofc, it depends on what you want to test, that expired tombstones are gone from the sstables? or from the system? the purge in row cache as you can see is lazy, since it happens eventually through reads.

Reply:
Yep, I think those were the partition tombstones we have seen remaining in the listing when we tried in the past. I tried with the "BYPASS CACHE" but it doesn't seem to work, the tombstones are still listed in that cache and drop caches is till needed to get rid of them

Reply:
> the purge in row cache as you can see is lazy, since it happens eventually through reads.

that was my reasoning. However, we do a read, and the expired tombstones appear there.

`bypass cache` probably applies to the virtual table that is created by `select ... from mutation_fragments(X)`, not to the underlying table `X`, hence it has no effect

Reply:
it's also possible that commitlog segment blocks tombstone GC, the age of the segment influences the calculation of "gc_before" (time point before which tombstones are considered expired). Perhaps a system wide flush (/storage_service/flush) will solve it.

Reply:
the mechanism you're describing would block tombstones stored in sstables from being compacted as well. In fact we have already dealt with and fixed this problem: issue: https://github.com/scylladb/scylladb/issues/20686, fix: https://github.com/scylladb/scylladb/pull/20691

what we see is that compaction gets rid of the tombstones, but reads show the tombstones that are still in cache